src/tasks/start/start-backend.ts

```diff
@@ -60,7 +62,7 @@ export async function startBackend(
 
   // Call preDao hook.
   if (hooks && hooks.preDao) {
-    await hooks.preDao(bre)
+    await hooks.preDao({}, bre)
```

Comment:
Introducing a consistent interface for all hooks: they now all have 2 parameters: 
1) params - can contain stuff like dao, proxy, etc.
2) bre - can contain stuff like web3, truffle artifacts, etc.

This is done by Buidler on its task interface and its pretty comfortable to know that you'll always have to deal with two objects.

Reply:
I'm a bit hesitant about naming it `params` 🤔

Reply:
I don't see the purpose of having a parameter typed `params: {}` that will always be an empty object. Consistency is not important anymore in this sense because typescript will enforce that you use `(params, bre)` or `(bre)` in each function accordingly

Reply:
Apps that use the plugin don't necessarily use Typescript. Having two fixed parameters is easier to reason about. Whenever you're writing a hook, you know that two things will come in. If you're using Typescript, great, you can use static types to understand what you need to do, but if you're not, you can inspect the first object and see what you've got. I've used this in Buidler's task interface and it's really handy. Also, this way, if we ever decide to add a parameter to a hook, like for example "accounts" in preDao(...), we don't break existing hooks. There's simply this extra params.accounts that people can access.

Reply:
Being future proof is a very good reason for keeping the API the same, agree!